pkg/controllers/workgenerator/controller.go

```diff
@@ -623,7 +655,7 @@ func (r *Reconciler) SetupWithManager(mgr controllerruntime.Manager) error {
 
 				// we only need to handle the case the applied or available condition is changed between the
 				// new and old work objects. Otherwise, it won't affect the binding applied condition
-				if condition.EqualCondition(oldAppliedStatus, newAppliedStatus) && condition.EqualCondition(oldAvailableStatus, newAvailableStatus) {
+				if condition.EqualCondition(newAppliedStatus, oldAppliedStatus) && condition.EqualCondition(newAvailableStatus, oldAvailableStatus) {
```

Comment:
we cannot change the order,

if newCondition.ObservedGeneration > oldCondition.ObservedGeneration, we need to requeue the request as the spec has been changed and we need to update other status fields.

For example, status keeps false but the failedManifest set are different.

Reply:
ha, the original order is wrong but now that we care about failedManifest, I am not sure if we can do this comparison anymore.

Reply:
we will keep this order and use a back off loop to requeue in the case that not all manifests are available.